ui/src/app/components/resource-card.tsx

```diff
@@ -112,71 +131,78 @@ function canDelete(resource: WorkspaceResource): boolean {
   return resource.permission === 'OWNER';
 }
 
-function getColor(resource: WorkspaceResource): string {
-  return fp.cond([
-    [isCohort, () => colors.resourceCardHighlights.cohort],
-    [isCohortReview, () => colors.resourceCardHighlights.cohortReview],
-    [isConceptSet, () => colors.resourceCardHighlights.conceptSet],
-    [isDataSet, () => colors.resourceCardHighlights.dataSet],
-    [isNotebook, () => colors.resourceCardHighlights.notebook],
-  ])(resource);
+interface NavProps extends PropsWithChildren<any> {
+  resource: WorkspaceResource;
+  linkTestId?: string;
+  style?: CSSProperties;
 }
 
+const ResourceNavigation = (props: NavProps) => {
+  const {resource, linkTestId, style = styles.resourceName, children} = props;
+  const url = getResourceUrl(resource);
+
+  function canNavigate(): boolean {
+    // can always navigate to notebooks
+    return isNotebook(resource) || canWrite(resource);
+  }
+
+  function onNavigate() {
+    if (isNotebook(resource)) {
+      AnalyticsTracker.Notebooks.Preview();
+    }
+  }
+
+  return <Clickable disabled={!canNavigate()}>
+    <a style={style}
+       data-test-id={linkTestId}
+       href={url}
+       onClick={e => {
+         onNavigate();
+         navigateAndPreventDefaultIfNoKeysPressed(e, url);
+       }}>
+      {...children}
+    </a>
+  </Clickable>;
+};
+
 interface Props {
   actions: Action[];
-  disabled: boolean;
   resource: WorkspaceResource;
-  onNavigate: () => void;
+  menuOnly: boolean;  // use this component strictly for its actions, without rendering the card
```

Comment:
Note: the full card view is still active on the Data and Analysis pages

Reply:
was this necessary for unit tests somehow?

Reply:
The leftmost column in the RecentResourcesTable is populated with ResourceCards with menuOnly = true, so only the snowman menu appears.  `data-page` and `notebooks-list` set menuOnly = false so they continue to render the full cards.

Reply:
Ah, I see. Would it make sense to pull the menu out and render it both here and inside the `ResourceCard`?

Reply:
Good point - no need for `ResourceCard` to have this prop if we choose not to render the ResourceCard at all when `menuOnly` is true.